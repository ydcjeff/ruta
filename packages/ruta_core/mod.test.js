import { create_routes, define_route } from './mod.js';
import { expect, test } from 'vitest';

test(create_routes.name, () => {
	const routes = create_routes()
		.add('', [
			define_route({
				path: '/',
				page: '/',
				error: '/',
			}),
		])
		.add('/', [
			define_route({
				path: '',
				page: '',
			}),
			define_route({
				path: 'child-1',
				page: 'child-1',
				error: 'child-1',
			}),
		])
		.add('/child-1', [
			define_route({
				path: 'grand-child-1',
				page: 'grand-child-1',
			}),
		])
		.add('/', [
			define_route({
				path: 'users',
				page: 'users',
			}),
		])
		.add('/users', [
			define_route({
				path: ':user_id',
				page: ':user_id',
			}),
			define_route({
				path: '',
				page: '',
			}),
			define_route({
				path: '/',
				page: '/',
			}),
		])
		.done();

	expect(routes).toMatchSnapshot();
});
