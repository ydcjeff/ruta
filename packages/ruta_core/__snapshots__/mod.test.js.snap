// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`create_routes 1`] = `
{
  "/": {
    "error": "/",
    "page": "/",
    "path": "/",
    Symbol(page): [
      "/",
      "",
    ],
    Symbol(error): [
      "/",
    ],
    Symbol(load): [],
    Symbol(parse_params): [],
    Symbol(parse_search): [],
    Symbol(has_layout): true,
  },
  "/child-1": {
    "error": "child-1",
    "page": "child-1",
    "path": "child-1",
    Symbol(page): [
      "/",
      "child-1",
    ],
    Symbol(error): [
      "child-1",
    ],
    Symbol(load): [],
    Symbol(parse_params): [],
    Symbol(parse_search): [],
  },
  "/child-1/grand-child-1": {
    "page": "grand-child-1",
    "path": "grand-child-1",
    Symbol(page): [
      "/",
      "child-1",
      "grand-child-1",
    ],
    Symbol(error): [
      "child-1",
    ],
    Symbol(load): [],
    Symbol(parse_params): [],
    Symbol(parse_search): [],
  },
  "/users": {
    "page": "users",
    "path": "users",
    Symbol(page): [
      "/",
      "users",
      "",
    ],
    Symbol(error): [],
    Symbol(load): [],
    Symbol(parse_params): [],
    Symbol(parse_search): [],
    Symbol(has_layout): true,
  },
  "/users/": {
    "page": "/",
    "path": "/",
    Symbol(page): [
      "/",
      "users",
      "/",
    ],
    Symbol(error): [],
    Symbol(load): [],
    Symbol(parse_params): [],
    Symbol(parse_search): [],
  },
  "/users/:user_id": {
    "page": ":user_id",
    "path": ":user_id",
    Symbol(page): [
      "/",
      "users",
      ":user_id",
    ],
    Symbol(error): [],
    Symbol(load): [],
    Symbol(parse_params): [],
    Symbol(parse_search): [],
  },
}
`;
